R2Northstar/NorthstarMods
# Arc Cannon charge effect fixes

Fixes the Arc Cannon's charge effect disappearing if you tried fanning it while using the "capacitor" mod, by making the effect turn off when its not charging. This is also how it worked in Titanfall 1.

Oddly enough, theres code thats seemingly *supposed* to do this in ``_arc_cannon.nut`` but it doesn't seem to work properly.

Before:

https://github.com/user-attachments/assets/82e138a7-1e29-433f-8dd9-ebfc1ce26ed5

After:

https://github.com/user-attachments/assets/8e014db0-95ac-4f6f-b18d-6cf55b2eae46

How to test:

1. Equip the Arc Cannon with the capacitor attachment (``give mp_weapon_arc_cannon capacitor``)
2. Start charging it up. 

Without this pr, if you stop charging the weapon, the effect wouldn't turn off. This also means that if you tried fanning it like you would with the charge rifle, the effect would eventually run for too long and disappear.

With this pr, the effect turns off when not charging as expected.

Edit: Also adds an extra unused charge effect. See below
